libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationCamera.kt

```diff
@@ -16,35 +14,43 @@ import com.mapbox.maps.extension.androidauto.DefaultMapboxCarMapGestureHandler
 import com.mapbox.maps.extension.androidauto.MapboxCarMapObserver
 import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
 import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.core.trip.session.LocationMatcherResult
 import com.mapbox.navigation.core.trip.session.LocationObserver
 import com.mapbox.navigation.core.trip.session.RouteProgressObserver
 import com.mapbox.navigation.ui.maps.camera.NavigationCamera
 import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
 import com.mapbox.navigation.ui.maps.camera.transition.NavigationCameraTransitionOptions
+import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponentContract
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 private const val DEFAULT_INITIAL_ZOOM = 15.0
 
 /**
  * Integrates the Android Auto [MapboxCarMapSurface] with the [NavigationCamera].
  */
-@OptIn(MapboxExperimental::class)
-class CarNavigationCamera internal constructor(
+@OptIn(MapboxExperimental::class, ExperimentalPreviewMapboxNavigationAPI::class)
+class CarNavigationCamera(
     private val mapboxNavigation: MapboxNavigation,
     private val initialCarCameraMode: CarCameraMode,
     private val alternativeCarCameraMode: CarCameraMode?,
-    private val routesProvider: RoutesProvider,
     private val initialCameraOptions: CameraOptions? = CameraOptions.Builder()
         .zoom(DEFAULT_INITIAL_ZOOM)
-        .build()
+        .build(),
+    private val contract: RouteLineComponentContract = MapboxRouteLineComponentContract(),
```

Comment:
This does not feel right.. But it's a simple way to share the route with the route line

Right now the `CameraComponent` from dropin-ui does not use a contract with shared route preview state.

Reply:
Am considering, bringing back the `RoutesProvider` interface because this contract does not make sense

Reply:
`RouteLineContract` should not be used in `Camera`. Unless `CameraComponent` in Drop-In is moved outside, AA should continue to use `RoutesProvider` for their own camera implementation.

Reply:
RoutesProvider is not making it simpler, adds an extra layer of glue